base/util.jl

```diff
@@ -750,3 +750,66 @@ kwdef_val(::Type{Cwstring}) = Cwstring(C_NULL)
 kwdef_val{T<:Integer}(::Type{T}) = zero(T)
 
 kwdef_val{T}(::Type{T}) = T()
+
+############################################################################
+# @views macro (not defined in subarray.jl because of a bootstrapping
+# issue with the code generation below).
+
+# maybeview is like getindex, but returns a view for slicing operations
+# (while remaining equivalent to getindex for scalar indices and non-array types)
+@propagate_inbounds maybeview(A, args...) = getindex(A, args...)
+@propagate_inbounds maybeview(A::AbstractArray, args...) = view(A, args...)
+@propagate_inbounds maybeview(A::AbstractArray, args::Number...) = getindex(A, args...)
+@propagate_inbounds maybeview(A) = getindex(A)
+@propagate_inbounds maybeview(A::AbstractArray) = getindex(A)
+# avoid splatting penalty in common cases:
```

Comment:
I tried a simple test case
```
f(x) = 1
f(x,y) = 2
g(x...) = f(x...)
```
and it wasn't getting inlined, but I guess I should try again with `@propagate_inbounds`?